### ISSUE TITLE
Support for esriFieldTypeDate

### ISSUE THREAD

**Issue by jfbourgon** [fgpv-vpgf/fgpv-vpgf#455]
Current version of RAMP is not interpreting numeric value defined as 'esriFieldTypeDate' and displays them as plain numeric value.

Here is an example of ESRI service using this datatype: 
http://services.aadnc-aandc.gc.ca/geomatics/rest/services/Donnees_Ouvertes-Open_Data/Bail_minier_NU_Mining_Lease/MapServer/0/query?where=OBJECTID%3E-1&outFields=*&returnGeometry=false&f=json&callback=dojo_request_script_callbacks.dojo_request_script10

Here is also a screenshot of how those fields are currently displayed in RAMP:
![image](https://cloud.githubusercontent.com/assets/1030226/14293680/a0f1d77c-fb3b-11e5-9547-1899a72aad4e.png)

As well as a screenshoot of the expected date format (as it is displayed in ArcGIS desktop)
![image](https://cloud.githubusercontent.com/assets/1030226/14294343/80424932-fb3e-11e5-8a61-ed387ac44d3b.png)

**Comment by mweech** [comment]
#374 must be completed first.

**Comment by james-rae** [comment]
Will also enhance the data grid so the conversion appears there too

**Comment by james-rae** [comment]
Decided to apply the conversion at the time of attribute load.  This way we don't have to pepper conversion code everywhere the fields are being accessed.  Logic will now lurk in geoApi attribute loader

**Comment by AleksueiR** [comment]
I think any styling on attributes should be done on the client as it's always be linked to the config if we want to allow for different styles (someone wants to display just the year for some reason). If you move the code to geoapi, it won't be possible.

To avoid multiple conversions in the client code, put it in the `formatAttributes` function in the `layer-registry` where it will run only once.

**Comment by alyec** [comment]
> To avoid multiple conversions in the client code, put it in the formatAttributes function in the layer-registry where it will run only once.

The other option would be to move the styling function to geoApi but allow the viewer to decide how it should be applied.

**Comment by james-rae** [comment]
`formatAttributes` formats stuff for grid consumption.  Other functions (identify, tooltips when implemented) will not be using the grid formatted store as data source.  

This also highlights another case I've missed -- identify results on dynamic layers are getting attributes directly from the server, so that is currently being missed, and having formatting in geoApi won't be helpful.  

Need to think a bit more, but sounds like we need a attribute formatter somewhere in the angular app that is easily accessed.  In particular, would like to cache specific things (e.g. don't want to constantly be examining the fields lists to check what fields are of date type).

My main concern right now is that this 2-estimate task is growing into a full-blown formatting framework.

**Comment by AleksueiR** [comment]
We can always format directly on the template with the angular date filter and in the table through the datatable render function.

It doesn't look like we can avoid constantly checking field types since you can't cache identify results from dynamic layers anyway.

**Comment by alyec** [comment]
> My main concern right now is that this 2-estimate task is growing into a full-blown formatting framework.

I'm good with leaving as is and making a TODO for future refactoring or going with Alex's format directly in template.

**Comment by james-rae** [comment]
After discussion with Mike, we will continue with short, specific fix for date only on client; get dates working and move on to higher priority items. A new issue (see above) has been created as a placeholder for bigger-picture formatting, and points to this discussion.
